### DISCUSSION TITLE
Cloud-Hosted Obsidian + local rest api plugin

### DISCUSSION THREAD

**Discussion by karljuhlthefool** [coddingtonbear/obsidian-local-rest-api#132] [Q&A] [answered]
I'm new to obsidian, but really appreciate your work on this great project! 

I'm looking into self hosting obsidian in the cloud and using livesync + couchdb to sync to my desktop/mobile obsidian versions. I wanted to ask if anyone has set up the local rest api plugin on the cloud hosted obsidian? 

I'm planning to self host obsidian and then have cron-triggered and webhook-triggered functions use the rest api on the cloud vm. Just wondering if someone else has done this kind of thing before I jump in.

**Comment by azdolinski** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/c418cd76-ba33-432f-beb0-83b43686de09)
This my setup which i made few months ago...  i also now using Omniserch plugin which allow search via API for images.

Obsidian work 24/7 in docker / protected by password and nginx... login via https open WebVNC [KasmVNC] so you can access and install all plugins...

```docker
services:
  obsidian:
    image: lscr.io/linuxserver/obsidian:latest
    container_name: obsidian
    hostname: obsidian
    security_opt:
      - seccomp:unconfined #optional
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Etc/UTC
      - CUSTOM_USER=admin
      - PASSWORD=${PASSWORD:-admin}
    volumes:
      - ./config:/config
    ports:
      - 7000:3000   # HTTP KasmVNC
      - 7001:3001   # HTTPS KasmVNC
      - 27123:27123   # Obsidian REST API HTTP
      - 27124:27124   # Obsidian REST API HTTPS
      - 51361:51361   # Obsidian REST HTTP Omnisearch
    shm_size: "1gb"
    mem_limit: 1G
    restart: unless-stopped
```

Stats with my 5 plugins for sync/rest/omnisearch and image processing:
```
CONTAINER ID   NAME       CPU %     MEM USAGE / LIMIT   MEM %     NET I/O           BLOCK I/O   PIDS
35b7fb6da7da   obsidian   0.30%     449.4MiB / 1GiB     43.89%    9.51MB / 6.69MB   0B / 0B     0
```

**Reply by karljuhlthefool** [reply]
Thanks so much for all the info!

**Reply by OleksandrKucherenko** [reply]
https://github.com/coddingtonbear/obsidian-local-rest-api/discussions/158

Another approach, i got more lightweight docker image
